pkg/cmd/dev/testlogic.go

```diff
@@ -297,6 +302,21 @@ func (d *dev) testlogic(cmd *cobra.Command, commandLine []string) error {
 	return nil
 }
 
+// This function determines if any test_logic or execbuilder/testdata files were
+// modified in the current branch, and if so, determines if we should re-generate logic tests.
+func (d *dev) shouldGenerateLogicTests(ctx context.Context) bool {
+	if buildutil.CrdbTestBuild {
+		return true
+	}
+	base, _ := d.getMergeBaseHash(ctx)
+	// Generate logic tests if the merge base hash isn't found
+	if base == "" {
+		return true
+	}
+	changedFiles, _ := d.exec.CommandContextSilent(ctx, "git", "diff", "--no-ext-diff", "--name-only", base, "--", "**/logic_test/**", "**/execbuilder/testdata/**")
```

Comment:
As mentioned in DM: use the list of files from `build/bazelutil/bazel-generate.sh` so we're consistent with it.

```
if files_unchanged_from_upstream WORKSPACE $(find_relevant ./pkg/sql/logictest/logictestbase -name BUILD.bazel -or -name '*.go') $(find_relevant ./pkg/sql/logictest/testdata -name '*') $(find_relevant ./pkg/sql/sqlitelogictest -name BUILD.bazel -or -name '*.go') $(find_relevant ./pkg/ccl/logictestccl/testdata -name '*') $(find_relevant pkg/sql/opt/exec/execbuilder/testdata -name '*'); then
```

I translate that to the following `git diff` command:

```
git diff ... -- pkg/sql/logictest/logictestbase/** pkg/sql/logictest/testdata/** pkg/sql/sqlitelogictest/BUILD.bazel pkg/sql/sqlitelogictest/sqlitelogictest.go pkg/ccl/logictestccl/testdata/** pkg/sql/opt/exec/execbuilder/testdata/**
```

Reply:
Oh sorry I was a little confused reading the DM -- thought you meant those files referred to the second option in avoiding failing unit tests. Fixed now.